src/main/kotlin/com/terraformation/backend/support/atlassian/resource/AtlassianResourceFactory.kt

```diff
@@ -0,0 +1,31 @@
+package com.terraformation.backend.support.atlassian.resource
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.support.atlassian.SupportRequestType
+import javax.inject.Named
+
+/** A factory class to inject configs or build information into the resource. */
+@Named
+class AtlassianResourceFactory(private val config: TerrawareServerConfig) {
```

Comment:
This is definitely nicer than the builder class.

What will the call sites look like? If it's always going to be the equivalent of `client.makeRequest(factory.createServiceDeskRequest(...))` like in the test cases, and the callers will never do anything with these objects other than immediately pass them to `AtlassianClient.makeRequest`, then would it make sense to turn this class into a service that actually sends the request and returns the response? Then callers don't need to have both this class and `AtlassianClient` as dependencies, and the request classes become internal implementation details that callers never need to see.

Reply:
This makes a lot of sense! I originally wanted to split up the service layer and the http client layer. The caller in this case would be a `SupportService` class. But I agree with you it just ends up complicating things. I will keep working on this PR to include permissions and use this as the service layer too.